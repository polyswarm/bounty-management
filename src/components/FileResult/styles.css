.FileResult-Indicator-Good {
  color: rgb(46, 255, 109);
}

.FileResult-Indicator-Warning {
  color: yellow;
}

.FileResult-Indicator-Bad {
  color: #ff1a00;
}
