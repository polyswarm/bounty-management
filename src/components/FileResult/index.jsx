import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Component Imports
import './styles.css';

class FileResult extends Component {

  render() {
    const { props: { children, good, total } } = this;
    const indicator = this.computeIndicator(good, total);
    return (
      <div className='FileResult'>
        {children}
        { indicator && (
          <p className={`FileResult-Indicator-${indicator}`}>{good}{' / '}{total}</p>
        )}
      </div>
    );
  }

  computeIndicator(good, total) {
    const ratio = good / total;
    if ( good == null || total == null || total == 0) {
      return null;
    }else if (ratio >= .7) {
      return 'Good';
    } else if (ratio >= .5) {
      return 'Warning';
    } else {
      return 'Bad';
    }
  }
}

FileResult.proptypes = {
  name: PropTypes.string.isRequired,
  good: PropTypes.number,
  total: PropTypes.number,
};
export default FileResult;
