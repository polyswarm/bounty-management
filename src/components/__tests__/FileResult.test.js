import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson, mountToJson} from 'enzyme-to-json';
import FileResult from '../FileResult';

it('renders without crashing', () => {
  const wrapper = render(<FileResult />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('returns Good when given 70 over 100', () => {
  expect(FileResult.computeIndicator(70, 100)).toEqual('Good');
});

it('returns Good when given 1000 over 100', () => {
  expect(FileResult.computeIndicator(1000, 100)).toEqual('Good');
});

it('returns Warning when given 699 over 1000 ', () => {
  expect(FileResult.computeIndicator(699, 1000)).toEqual('Warning');
});

it('returns Bad when given 49 over 100 ', () => {
  expect(FileResult.computeIndicator(49, 100)).toEqual('Bad');
});

it('returns Bad when given 0 over 1', () => {
  expect(FileResult.computeIndicator(0, 1)).toEqual('Bad');
});

it('returns Bad when given a negative number', () => {
  expect(FileResult.computeIndicator(-1, 10)).toEqual('Bad');
  expect(FileResult.computeIndicator(10, -1)).toEqual('Bad');
});

it('returns null when given 1 over 0', () => {
  expect(FileResult.computeIndicator(1, 0)).toEqual(null);
});

it('doesnt show any Indicator if good isn\'t used', () => {
  const wrapper = mount(<FileResult total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Good')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Warning')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Bad')).toHaveLength(0);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('doesnt show any Indicator if total isn\'t used', () => {
  const wrapper = mount(<FileResult good={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Good')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Warning')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Bad')).toHaveLength(0);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('doesnt show any Indicator if good is null', () => {
  const wrapper = mount(<FileResult good={null} total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Good')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Warning')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Bad')).toHaveLength(0);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('doesnt show any Indicator if total is null', () => {
  const wrapper = mount(<FileResult good={10} total={null}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Good')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Warning')).toHaveLength(0);
  expect(wrapper.find('.FileResult-Indicator-Bad')).toHaveLength(0);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('shows good when ratio > 70%', () => {
  const wrapper = mount(<FileResult good={70} total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Good')).toHaveLength(1);
  expect(wrapper.find('.FileResult-Indicator-Good').text()).toEqual('70 / 100');
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('shows warning when ratio at 69%', () => {
  const wrapper = mount(<FileResult good={69} total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Warning')).toHaveLength(1);
  expect(wrapper.find('.FileResult-Indicator-Warning').text()).toEqual('69 / 100');
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('shows warning when ratio at 50%', () => {
  const wrapper = mount(<FileResult good={50} total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Warning')).toHaveLength(1);
  expect(wrapper.find('.FileResult-Indicator-Warning').text()).toEqual('50 / 100');
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('shows bad when ratio at 49%', () => {
  const wrapper = mount(<FileResult good={49} total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Bad')).toHaveLength(1);
  expect(wrapper.find('.FileResult-Indicator-Bad').text()).toEqual('49 / 100');
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('shows bad when ratio is 0%', () => {
  const wrapper = mount(<FileResult good={0} total={100}>asdf</FileResult>);

  expect(wrapper.find('.FileResult-Indicator-Bad')).toHaveLength(1);
  expect(wrapper.find('.FileResult-Indicator-Bad').text()).toEqual('0 / 100');
  expect(mountToJson(wrapper)).toMatchSnapshot();
});
