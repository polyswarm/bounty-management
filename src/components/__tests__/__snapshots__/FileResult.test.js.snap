// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnt show any Indicator if good is null 1`] = `
<FileResult
  good={null}
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
  </div>
</FileResult>
`;

exports[`doesnt show any Indicator if good isn't used 1`] = `
<FileResult
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
  </div>
</FileResult>
`;

exports[`doesnt show any Indicator if total is null 1`] = `
<FileResult
  good={10}
  total={null}
>
  <div
    className="FileResult"
  >
    asdf
  </div>
</FileResult>
`;

exports[`doesnt show any Indicator if total isn't used 1`] = `
<FileResult
  good={100}
>
  <div
    className="FileResult"
  >
    asdf
  </div>
</FileResult>
`;

exports[`renders without crashing 1`] = `
<div
  class="FileResult"
>
  
</div>
`;

exports[`shows bad when ratio at 49% 1`] = `
<FileResult
  good={49}
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
    <p
      className="FileResult-Indicator-Bad"
    >
      49
       / 
      100
    </p>
  </div>
</FileResult>
`;

exports[`shows bad when ratio is 0% 1`] = `
<FileResult
  good={0}
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
    <p
      className="FileResult-Indicator-Bad"
    >
      0
       / 
      100
    </p>
  </div>
</FileResult>
`;

exports[`shows good when ratio > 70% 1`] = `
<FileResult
  good={70}
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
    <p
      className="FileResult-Indicator-Good"
    >
      70
       / 
      100
    </p>
  </div>
</FileResult>
`;

exports[`shows warning when ratio at 50% 1`] = `
<FileResult
  good={50}
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
    <p
      className="FileResult-Indicator-Warning"
    >
      50
       / 
      100
    </p>
  </div>
</FileResult>
`;

exports[`shows warning when ratio at 69% 1`] = `
<FileResult
  good={69}
  total={100}
>
  <div
    className="FileResult"
  >
    asdf
    <p
      className="FileResult-Indicator-Warning"
    >
      69
       / 
      100
    </p>
  </div>
</FileResult>
`;
